### PR TITLE
Cloud 0.23.5 release notes

### PR DESCRIPTION
**Pull request by paolaperaza**
Initializing this PR to capture changes that we expect to ship with Astronomer Cloud v0.23.4.

Basing changes on this milestone: https://github.com/astronomer/issues/issues?q=is%3Aopen+is%3Aissue+milestone%3A0.23.4

Still a draft for now while I finish up.

### CODE REVIEW

**Comment by paolaperaza** [comment]
Alright, @vishwas-astro ! I just made some last-minute changes to this PR to reflect the latest v0.23.5 additions and remove support for Webserver reboot, since that is no longer in scope.

Can you take a look and let me know if I'm missing anything/you have any corrections?

File: cloud/stable/06_resources/01_release-notes.md [hunk 1]
```diff
@@ -4,13 +4,55 @@ navTitle: "Release Notes"
...
+
+Airflow 1.10.14 was built to make testing and migration to Airflow 2.0 as easy as possible. Highlights include:
+
+- "Warning" to users with duplicate Airflow Connections ([commit](https://github.com/apache/airflow/commit/0e40ddd8e))
+- Enable [DAG Serialization](https://airflow.apache.org/docs/apache-airflow/stable/dag-serialization.html) by default ([commit](https://github.com/apache/airflow/commit/8a265067e))
+- Support for Airflow 2.0 CLI commands ([commit](https://github.com/apache/airflow/pull/12725))
+- Bugfix: Unable to import Airflow plugins on Python 3.8 ([commit](https://github.com/apache/airflow/pull/12859))
+- BugFix: Tasks with depends_on_past or task_concurrency are stuck ([commit](https://github.com/apache/airflow/pull/12663))
+
+To upgrade to AC 1.10.14, add our new Debian image to your Dockerfile:
```
Comment (vishwas-astro) [inline cloud/stable/06_resources/01_release-notes.md:25]: Do we want to include that there are no more alpine images.

File: cloud/stable/06_resources/01_release-notes.md [hunk 2]
```diff
@@ -4,13 +4,55 @@ navTitle: "Release Notes"
...
+As of Astronomer v0.23.5, we:
+
+- Enforce that all users migrate to Astronomer Certified 1.10.14 before upgrading to 2.0
+- Support Airflow's ["upgrade check"](http://apache-airflow-docs.s3-website.eu-central-1.amazonaws.com/docs/apache-airflow/latest/upgrade-check.html) in the Astronomer CLI (`$ astro dev upgrade-check`)
+
+The migration path to Airflow 2.0 on Astronomer follows all recommendations published by the Apache Airflow project. We expect to publish an official "Upgrade to Airflow 2.0 on Astronomer" guide as soon as Airflow 2.0 is generally available.
+
+#### Bug Fixes & Improvements
+
+- Add clear messaging to Astronomer UI aroundss steps required to finalize Airflow upgrade
```
Comment (vishwas-astro) [inline cloud/stable/06_resources/01_release-notes.md:46]: typo - aroundss?

**Review by vishwas-astro — APPROVED** [review]

File: cloud/stable/06_resources/01_release-notes.md (hunk 2 above)
Comment (paolaperaza) [reply cloud/stable/06_resources/01_release-notes.md]: Yikes, good catch

File: cloud/stable/06_resources/01_release-notes.md (hunk 1 above)
Comment (paolaperaza) [reply cloud/stable/06_resources/01_release-notes.md:25]: Yes, good call - added!

**Pull request merged**